pkg/drain/nodegroup.go

```diff
@@ -144,18 +145,31 @@ func NodeGroup(clientSet kubernetes.Interface, ng *api.NodeGroup, waitTimeout ti
 
 			for _, node := range nodes.Items {
 				if newPendingNodes.Has(node.Name) {
+					retry := 1
 					pending, err := evictPods(drainer, &node)
-					if err != nil {
-						return err
-					}
-					logger.Debug("%d pods to be evicted from %s", pending, node.Name)
-					if pending == 0 {
-						drainedNodes.Insert(node.Name)
+					for pending > 0 {
+						logger.Debug("%d pods to be evicted from %s", pending, node.Name)
+						time.Sleep(5 * time.Second)
+						pending, err = evictPods(drainer, &node)
+						if err != nil && retry < 60 {
```

Comment:
Please, move the 60 to a constant with a name that can be searched later on if we need to like "podEvictionMaxRetries".